crypto/src/asset.rs

```diff
@@ -30,19 +33,24 @@ pub struct Id(pub Fq);
 
 pub struct Denom(pub String);
 
+impl Id {
+    pub fn to_bech32(&self) -> String {
+        use ark_ff::BigInteger;
+        let bytes = self.0.into_repr().to_bytes_le().to_base32();
+        bech32::encode(PENUMBRA_BECH32_ASSET_PREFIX, bytes, Variant::Bech32m)
+            .expect("bech32 hrp is valid")
+    }
+}
+
 impl std::fmt::Debug for Id {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        use ark_ff::BigInteger;
-        let bytes = self.0.into_repr().to_bytes_le();
-        f.write_fmt(format_args!("asset::Id({})", hex::encode(&bytes)))
+        f.write_fmt(format_args!("asset::ID({})", self.to_bech32()))
```

Comment:
Do we want debug to have the same output as Display? The only difference as of this PR is that debug will print `asset::ID(...bech32 addr...)` instead of just `bech32 addr`. This seems potentially helpful so we can distinguish in logging, but the HRP should provide that for us as well.

Reply:
Hmm, I think the HRP probably tells us what the type is, and keeping the debug output compact seems nice.